src/mumble/UserLocalVolumeDialog.cpp

```diff
@@ -99,8 +101,11 @@ void UserLocalVolumeDialog::on_qbbUserLocalVolume_clicked(QAbstractButton *butto
 	}
 	if (button == qbbUserLocalVolume->button(QDialogButtonBox::Ok)) {
 		ClientUser *user = ClientUser::get(m_clientSession);
-		if (user && !user->qsHash.isEmpty()) {
-			g.db->setUserLocalVolume(user->qsHash, user->fLocalVolume);
+		if (user) {
+			if (!user->qsHash.isEmpty())
+			    g.db->setUserLocalVolume(user->qsHash, user->fLocalVolume);
+			else
+			    g.mw->msgBox(tr("This change could not be saved permanently and is lost on restart because %1 does not have a certificate.").arg(Log::formatClientUser(user, Log::Target)));
```

Comment:
Instead of outputting the message here and having to include `MainWindow.h` and `Log.h` this message could also be displayed in `MainWindow.cpp`. However that would show the before the change was made and even if the local volume adjustment is cancelled. Might be worth though for an egde-case as small as this.

Reply:
Including these header files here shouldn't matter. I don't even think that it noticeably slows down the compilation. I prefer logging the error when it actually happens and not when it will potentially happen